src/main/java/me/lokka30/phantomworlds/commands/phantomworlds/subcommands/TeleportSubcommand.java

```diff
@@ -38,68 +45,47 @@
-
-        if (Bukkit.getWorld(args[1]) == null) {
-            (new MultiMessage(
-                    main.messages.getConfig().getStringList("command.phantomworlds.subcommands.common.invalid-world"), Arrays.asList(
-                    new MultiMessage.Placeholder("prefix", main.messages.getConfig().getString("common.prefix", "&b&lPhantomWorlds: &7"), true),
-                    new MultiMessage.Placeholder("world", args[1], false)
-            ))).send(sender);
-            return;
-        }
-
-        Player target;
-        if (args.length == 3) {
-            target = Bukkit.getPlayer(args[2]);
-
-            // If the target is offline or invisible to the sender, then stop
-            if (target == null || !Utils.getPlayersCanSeeList(sender).contains(target.getName())) {
-                (new MultiMessage(
-                        main.messages.getConfig().getStringList("command.phantomworlds.subcommands.common.player-offline"), Arrays.asList(
-                        new MultiMessage.Placeholder("prefix", main.messages.getConfig().getString("common.prefix", "&b&lPhantomWorlds: &7"), true),
-                        new MultiMessage.Placeholder("player", args[2], false)
-                ))).send(sender);
-                return;
-            }
-        } else {
-            if (sender instanceof Player) {
-                target = (Player) sender;
-            } else {
-                (new MultiMessage(
-                        main.messages.getConfig().getStringList("command.phantomworlds.subcommands.teleport.usage-console"), Arrays.asList(
-                        new MultiMessage.Placeholder("prefix", main.messages.getConfig().getString("common.prefix", "&b&lPhantomWorlds: &7"), true),
-                        new MultiMessage.Placeholder("label", label, false)
-                ))).send(sender);
-                return;
-            }
-        }
-
-        //noinspection ConstantConditions
-        target.teleport(Bukkit.getWorld(args[1]).getSpawnLocation());
-
-        (new MultiMessage(
-                main.messages.getConfig().getStringList("command.phantomworlds.subcommands.teleport.success"), Arrays.asList(
-                new MultiMessage.Placeholder("prefix", main.messages.getConfig().getString("common.prefix", "&b&lPhantomWorlds: &7"), true),
-                new MultiMessage.Placeholder("player", target.getName(), false),
-                new MultiMessage.Placeholder("world", args[1], false)
-        ))).send(sender);
+    	
+    	switch(subCommand) {
+    		case "teleport":
+    			teleportToWorld(main, sender, subCommand, label, args[2], args[1]);
+    			break;
+    		case "spawn":
+    			String world;
+    			if(args.length != 2) {
+    				world = Bukkit.getPlayer(sender.getName()).getWorld().toString();
+    			} else {
+    				world = Bukkit.getPlayer(args[1]).getWorld().toString();
+    			}
+    			teleportToWorld(main, sender, subCommand, label, args[1], world);
+    			break;
+    		default:
+    			return;	
```

Comment:
Comment to self: Instead of `return;`, `throw new IllegalStateException("Unexpected value" + subcommand);`